apps/src/StudioApp.js

```diff
@@ -2682,7 +2682,8 @@ StudioApp.prototype.handleUsingBlockly_ = function(config) {
       false
     ),
     useModalFunctionEditor: utils.valueOr(
-      config.level.useModalFunctionEditor,
+      config.level.edit_blocks !== TOOLBOX_EDIT_MODE &&
```

Comment:
This seems overcomplicated? Is `config.level.useModalFunctionEditor` a boolean? If so, then isn't this equivalent to just `useModalFunctionEditor: config.level.edit_blocks !== TOOLBOX_EDIT_MODE && config.level.useModalFunctionEditor` (no need for `utils.valueOr`)?

Reply:
I believe the reason for using `utils.valueOr` is that config.level.useModalFunctionEditor can be undefined whereas we always want useModalFunctionEditor to be T/F.

Reply:
Then I think you want like, `edit_blocks !== TOOLBOX && !!useModalFunctionEditor`
To me it seems like a little bit of a code smell to use `valueOr` with a boolean as the default value, because the boolean expression can almost certainly be simplified.

Reply:
👍 (thanks for the offline conversation) Also added a jira ticket with more details about future plans! https://codedotorg.atlassian.net/browse/STAR-944